app-modules/consent/graphql/consent-agreement.graphql

```diff
@@ -0,0 +1,167 @@
+#
+#    "The body of the consent agreement."
+#    body: String!
+#        @rules(apply: ["required", "string", "max:65535"])
+#}
+
+input UpdateConsentAgreementInput {
+    #    "The type of the consent agreement."
+    #    type: ConsentAgreementType
+    #        @rules(apply: ["nullable"])
+
+    "The title of the consent agreement."
+    title: String @rules(apply: ["nullable", "string", "max:255"])
+
+    "The description of the consent agreement."
+    description: String @rules(apply: ["nullable", "string", "max:65535"])
+
+    "The body of the consent agreement."
+    body: String @rules(apply: ["nullable", "string", "max:65535"])
+
+    users: UpdateUserConsentAgreementBelongsToManyInput
+}
+
+input UpdateUserConsentAgreementBelongsToManyInput {
+    consent: [ConnectUserConsentAgreementInput!] @rename(attribute: "connect")
+    #    revoke: [UUID!] @rename(attribute: "disconnect")
+}
+
+input ConnectUserConsentAgreementInput {
+    id: UUID! @rules(apply: ["required", "uuid", "exists:users,id"])
+    ip_address: String! @rules(apply: ["required", "string", "ip"])
+}
+
+type ConsentAgreementMutations {
+    #    "Create a new consent agreement."
+    #    create(
+    #        "The attributes to create a consent agreement."
+    #        input: CreateConsentAgreementInput! @spread
+    #    ): ConsentAgreement! @create @canModel(ability: "create")
+
+    "Update an existing consent agreement."
+    update(
+        "The identifier of the consent agreement you would like to update."
+        id: UUID!
+            @whereKey
+            @rules(apply: ["required", "uuid", "exists:consent_agreements"])
+
+        "The attributes to update a consent agreement."
+        input: UpdateConsentAgreementInput! @spread
+    ): ConsentAgreement! @update @canFind(ability: "update", find: "id")
+
+    #    "Delete an existing consent agreement."
+    #    delete(
+    #        "The identifier of the consent agreement you would like to delete."
+    #        id: UUID!
+    #            @rules(apply: ["required", "uuid", "exists:consent_agreements"])
+    #    ): ConsentAgreement! @delete @canFind(ability: "delete", find: "id")
+
+    consentUserToAgreement(
+        id: UUID!
+            @whereKey
+            @rules(apply: ["required", "uuid", "exists:consent_agreements"])
+        user_id: UUID! @rules(apply: ["required", "uuid", "exists:users,id"])
+    ): ConsentAgreement!
+        @canFind(ability: "update", find: "id")
+        @field(resolver: "AdvisingApp\\Consent\\GraphQL\\Mutations\\Consent")
+
+    #    revokeUserConsentFromAgreement(
+    #        id: UUID!
+    #            @whereKey
+    #            @rules(apply: ["required", "uuid", "exists:consent_agreements"])
+    #        user_id: UUID!
+    #            @rules(apply: ["required", "uuid", "exists:users,id"])
+    #    ): ConsentAgreement! @canFind(ability: "update", find: "id") @field(resolver: "AdvisingApp\\Consent\\GraphQL\\Mutations\\Revoke")
```

Comment:
Does consent and revoke belong here?

Because isn't consent simply dictated by the presence of a `UserConsentAgreement` record for the User?

So therefore consent and revoke is the created and delete Action of UserConsentAgreement?

Reply:
I removed the `UserConsentAgreement` mutations because there's nothing that should be updated on the record. Deletion is not supported, and creation should be handle by the parent `ConsentAgreement` record. In general you should not be manipulating the pivot record directly. Additionally, I left the `consent` mutation on the `ConsentAgreement` schema because one of the philosophies of GraphQL is to go beyond Rest and provide convenience and I believe calling this mutation is better than touching the pivot record.